pkg/controller/autoscaler/horizontalpodautoscaler_controller.go

```diff
@@ -113,7 +141,89 @@ func (a *HorizontalPodAutoscalerController) reconcileAutoscalers() error {
 			continue
 		}
 
+		var metrics heapster.MetricResultList
+		err = json.Unmarshal(resultRaw, &metrics)
+		if err != nil {
+			glog.Warningf("Failed to unmarshall heapster response: %v", err)
+			continue
+		}
+
 		glog.Infof("Metrics available for %s: %s", reference, string(resultRaw))
+
+		currentConsumption, count := metricSpec.aggregator(metrics)
+		if count != len(podList.Items) {
+			glog.Warningf("Metrics obtained for %d/%d of pods", count, len(podList.Items))
+			continue
+		}
+
+		// if the ratio is 1.2 we want to have 2 replicas
+		desiredReplicas := 1 + int((currentConsumption.Quantity.MilliValue()*int64(count))/hpa.Spec.Target.Quantity.MilliValue())
```

Comment:
In the design proposal, there were some ideas to take floor for scaling up and ceiling for scaling down, to avoid trashing. Please, just add TODO to update this formula.